### PR TITLE
[Azure Logs] Add caller_ip_address field for sign-in logs

### LINKED ISSUE #8829: [Azure Logs] Add missing fields in pipeline processing for sign-in logs
In [beats issue #37249](https://github.com/elastic/beats/issues/37249), it was discovered that some fields are not being processed correctly for Azure sign-in logs.

For example `callerIpAddress` is automatically being renamed in `caller_ip_address` but it is not processed in the expected json.

### PR DESCRIPTION
**Pull request by lucian-ioan**
## Proposed commit message
Add pipeline processing for callerIpAddress field to the Azure sign-in logs.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

- [ ] 

## Related issues

-  Closes #8829

### CODE REVIEW

**Comment by lucian-ioan** [comment]
Checking all other fields like `tenant_id` etc. but as far as I can tell those are being processed.

File: packages/azure/data_stream/signinlogs/elasticsearch/ingest_pipeline/default.yml [hunk 1]
```diff
@@ -83,8 +83,7 @@ processors:
       value: '{{{azure.signinlogs.properties.ip_address}}}'
       ignore_empty_value: true
   - set:
-      if: ctx?.source?.address == null
-      field: source.address
+      field: azure.signinlogs.properties.caller_ip_address
```
Comment (muthu-mps) [inline packages/azure/data_stream/signinlogs/elasticsearch/ingest_pipeline/default.yml:86]: When we add the `signinlogs.caller_ip_address` into the `signinlogs.properties.caller_ip_address` what happens with the following use-cases, 
1) The `callerIpAddress` field is outside the properties when we receive from the Azure. Can we capture this field in the same way?
2) In a certain scenarios we will get `caller_ip_address` inside the properties as well. How do we handle that?

Comment (lucian-ioan) [reply packages/azure/data_stream/signinlogs/elasticsearch/ingest_pipeline/default.yml]: 1. Done, makes sense to keep it as the original.
2. I couldn't find a scenario where `caller_up_address` is inside the properties unless dynamic mapping was applied, which shouldn't be a problem anymore.

File: packages/azure/data_stream/signinlogs/elasticsearch/ingest_pipeline/default.yml [hunk 2]
```diff
@@ -93,11 +92,15 @@ processors:
       type: ip
       ignore_missing: true
       ignore_failure: true
+  - convert:
+      field: azure.signinlogs.properties.caller_ip_address
```
Comment (muthu-mps) [inline packages/azure/data_stream/signinlogs/elasticsearch/ingest_pipeline/default.yml:96]:
```suggestion
      field: azure.signinlogs.caller_ip_address
```

File: packages/azure/data_stream/signinlogs/elasticsearch/ingest_pipeline/default.yml [hunk 3]
```diff
@@ -83,8 +83,7 @@ processors:
       value: '{{{azure.signinlogs.properties.ip_address}}}'
       ignore_empty_value: true
   - set:
-      if: ctx?.source?.address == null
-      field: source.address
+      field: azure.signinlogs.caller_ip_address
```
Comment (bhapas) [inline packages/azure/data_stream/signinlogs/elasticsearch/ingest_pipeline/default.yml:86]: What is this `set` doing? Looks like it is just picking up the value from same field?

Comment (lucian-ioan) [reply packages/azure/data_stream/signinlogs/elasticsearch/ingest_pipeline/default.yml]: Correct, it's redundant now.

**Review by bhapas — APPROVED** [review]

**Review by muthu-mps — APPROVED** [review]
LGTM!

**Pull request merged**